### PR TITLE
[Ray] Implements get_chunks_result for Ray execution context

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?

- [x] Implement  the`get_chunks_result` API for Ray execution context.
- [x] Fix remote object may be called before creation.
- [x] Add ray_dag CI, then we can use `@pytest.mark.ray_dag` to mark the test case run in ray backend.

## Related issue number

https://github.com/mars-project/mars/issues/2893

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/services/task/execution/ray/executor.py [hunk 1]
```diff
@@ -235,32 +245,33 @@ async def execute_subtask_graph(
...
-            output_object_refs.append(object_ref)
+            object_ref = task_context[chunk_key]
+            output_object_refs.add(object_ref)
             chunk_meta = key_to_meta.get(chunk_key)
             if chunk_meta is not None:
                 chunk_to_meta[chunk] = ExecutionChunkResult(chunk_meta, object_ref)
 
         logger.info("Waiting for stage %s complete.", stage_id)
-        ray.wait(output_object_refs, fetch_local=False)
+        ray.wait(list(output_object_refs), fetch_local=False)
```
Comment (chaokunyang) [inline mars/services/task/execution/ray/executor.py:274]: Will `rat.wait` block event loop? Can we invoke it with `to_thread`?

Comment (fyrestone) [reply mars/services/task/execution/ray/executor.py:274]: Good suggestion. Thanks.

File: mars/services/task/execution/ray/executor.py [hunk 2]
```diff
@@ -104,6 +104,9 @@ def execute_subtask(
     return output_values[0] if len(output_values) == 1 else output_values
 
 
+_ray_executor = ray.remote(execute_subtask) if ray else None
```
[the remark this answers is not in the document]
Comment (fyrestone) [reply mars/services/task/execution/ray/executor.py]: Fixed. Thanks.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**